Sources/Sentry/SentryProfiler.mm

```diff
@@ -262,6 +266,7 @@ + (void)timeoutAbort
 
     SENTRY_LOG_DEBUG(@"Stopping profiler %@ due to timeout.", _gCurrentProfiler);
     [self stopProfilerForReason:SentryProfilerTruncationReasonTimeout];
+    [_gCurrentProfiler captureEnvelope];
```

Comment:
Is this consistent with the transaction behavior? Do we capture transaction envelopes in these cases too?

Reply:
This is one of the places where a transaction might get discarded due to exceeding the hardcoded maximum time limit, I think. So we’ll either need to keep those transactions and just truncate them, or wait to upload profiles until those transactions are actually finished/kept.